test/setup/cluster_setup.go

```diff
@@ -279,6 +280,56 @@ func (c *Cluster) StartServer(s *Server) error {
 	return s.start(c.cmdTimeout)
 }
 
+func (c *Cluster) GetServerByID(serverID string) (*Server, int) {
+	if serverID == "" {
+		return nil, -1
+	}
+	for i, srv := range c.Servers {
+		if srv.serverID == serverID {
+			return srv, i
+		}
+	}
+	return nil, -1
+}
+
+func (c *Cluster) AgreedLeader(t *testing.T, activeServers ...int) int {
+	var leaders [3]int
```

Comment:
This assumes there will only be 3 servers, which is not the case. the size is the length of the `c.Servers`, so use a slice

Reply:
done